app/views/products/index.html.haml

```diff
@@ -1,26 +1,19 @@
 .d-grid.gap-2.d-md-block.mx-5
-  %button.btn.btn-dark.me-2.position-relative(type="button")= link_to 'New Category', "#", class: 'text-light text-decoration-none stretched-link'
   %button.btn.btn-dark.me-2.position-relative(type="button")= link_to 'New Product', new_product_path, class: 'text-light text-decoration-none stretched-link'
 
 #products-list-container.d-flex.my-2.mx-5.bg-light.position-relative
   %nav#products-list-navigation.d-flex.flex-column.align-items-start.p-2
-    %a.list-group-item.list-group-item-action(href="#category-1") Category 1
-    %a.list-group-item.list-group-item-action(href="#category-2") Category 2
-    %a.list-group-item.list-group-item-action(href="#category-3") Category 3
-    %a.list-group-item.list-group-item-action(href="#category-4") Category 4
-  #products-list.scrollspy.flex-grow-1.p-3(data-bs-offset="100" data-bs-spy="scroll" data-bs-target="#products-list-navigation" tabindex="0")
-    %h4#category-1 Category 1
-    - @products.each do |product|
-      .d-flex.align-items-center.list-group-item.list-group-item-action.p-1
-        .px-2= link_to product.name, product, class: 'text-dark text-decoration-none'
-        .px-2.flex-grow-1.text-align-start= "(#{product.unit})"
-        .px-2.mx-2.edit-icon.shc-icon.position-relative= link_to "", edit_product_path(product), class: "stretched-link"
-        .px-2.mx-2.delete-icon.shc-icon.position-relative= link_to "", product, data: { turbo_method: :delete, turbo_confirm: "Are you sure?" }, class: "stretched-link"
-    %h4#category-2 Category 2
-    %p ...
-    %h4#category-3 Category 3
-    %p ...
-    %h4#category-4 Category 4
-    %p ...
+    - @product_types.each do |type|
+      %a.list-group-item.list-group-item-action.p-1{:href => "#" + product_type_haml_id(type)}= type.titleize
+
+  #products-list.scrollspy.flex-grow-1.p-3(data-bs-offset="0" data-bs-spy="scroll" data-bs-target="#products-list-navigation" tabindex="0")
+    - @product_types.each do |type_name|
```

Comment:
You might consider using Enumerable#group_by method here for grouping products by their types. This way you will get all products already in one hash and you could iterate over this hash here.